core/chains/evm/chain.go

```diff
@@ -250,7 +250,16 @@ func newChain(ctx context.Context, cfg *evmconfig.ChainScoped, nodes []*toml.Nod
 		if opts.GenLogPoller != nil {
 			logPoller = opts.GenLogPoller(chainID)
 		} else {
-			logPoller = logpoller.NewObservedLogPoller(logpoller.NewORM(chainID, db, l, cfg.Database()), client, l, cfg.EVM().LogPollInterval(), int64(cfg.EVM().FinalityDepth()), int64(cfg.EVM().LogBackfillBatchSize()), int64(cfg.EVM().RPCDefaultBatchSize()), int64(cfg.EVM().LogKeepBlocksDepth()))
+			logPoller = logpoller.NewObservedLogPoller(
+				logpoller.NewORM(chainID, db, l, cfg.Database()),
+				client,
+				l,
+				cfg.EVM().LogPollInterval(),
+				cfg.EVM().FinalityTagEnabled(),
+				int64(cfg.EVM().FinalityDepth()),
```

Comment:
I'm guessing it's because when we do arithmetic on the blocks which are 64-bit we get a result that's 64-bits, so it would be awkward to force-cast that into 32-bits to make it match the finality depth size defined in the toml config. I guess we could expand the finality depth in the toml config to be 64 bits, but there's definitely no reason we'd ever need a finality depth larger than 2^32... I think casting here to 64-bit is okay.